src/lighteval/metrics/metrics_corpus.py

```diff
@@ -111,6 +111,6 @@ def compute(self, items: list[PerplexityCorpusMetricInput]):
         if self.metric_type == "perplexity":
             return math.exp(-np.mean(logprobs))
         if self.metric_type == "weighted_perplexity":
-            return math.exp(-np.average(logprobs, weights=weights))
+            return math.exp(-sum(logprobs) / sum(weights))
         if self.metric_type == "bits_per_byte":
-            return -np.average(logprobs, weights=weights) / math.log(2)
+            return -sum(logprobs) / sum(weights) * 1 / math.log(2)
```

Comment:
why are we changing this ?

Reply:
The logprobs we get are summed over each sample - here, when we aggregate over all samples, we want to get the overall perplexity of the full dataset, aka the sum of logprobs over the number of items (tokens or bytes).
`np.average` is doing `(logprob1*weight1+logprob2*weight2+...)/sum(weights)`